blog/2024-04-30-introducing-opencost-plugins/index.mdx

```diff
@@ -0,0 +1,153 @@
+
+![OpenCost](./img/opencost.png)
+
+The plugin framework for adding additional cost sources to OpenCost is flexible enough to allow for cost monitoring from a wide range of tools and services – it supports new costs that conform to the [FOCUS spec](https://focus.finops.org/) developed by the FinOps Foundation to ensure costs are shown in a consistent way. If you are interested in a plugin to monitor your resources, you can review the process for [adding a new plugin](https://github.com/opencost/opencost-plugins?tab=readme-ov-file#creating-a-new-plugin) and make a contribution, or [open an issue](https://github.com/opencost/opencost-plugins/issues) requesting a new plugin. Any contributor who adds a new plugin to OpenCost will receive a box full of OpenCost swag and $1000! If you have any issues or questions you can [join our slack community](https://slack.cncf.io/) or email us at [opencost@kubecost.com](mailto:opencost@kubecost.com). 
+
+To kick things off, we have created the first OpenCost plugin for Datadog cost monitoring! OpenCost 1.110 users are now able to monitor what it costs them to run Datadog’s various services in real time right alongside their monitoring of costs from Kubernetes and Cloud Services. Simply [install the Datadog plugin](https://www.opencost.io/docs/integrations/datadog), and you are able to slice and dice your Datadog costs however you want – by Account, Resource Type, Resource Name, etc. This granular, real-time visibility makes it easy to understand your costs by business unit (team, application, etc.), quickly identify increases in spend, and ensure you are spending money efficiently.   
+
+Below you can see how the new OpenCost API’s can be used to view costs from the configured plugins. For example, once you’ve [integrated your Datadog account](https://www.opencost.io/docs/integrations/datadog), you can access Datadog cost information via the following APIs:
+
+* customCost/total
+    * Used to grab a summary of custom costs over a window
+    * Available aggregations:
+        * zone
+        * accountName
+        * chargeCategory
+        * description
+        * resourceName
+        * resourceType
+        * providerId
+        * usageUnit
+        * domain
+        * costSource
+    * The available filters are the same as the available aggregations. For example:
+        * filter=domain:”datadog”
+        * filter=zone:”us”
+        * filter=resourceType:”infra_hosts”
+* customCost/timeseries
+    * Essentially equivalent to calling /total over a range of time steps. For example, querying for the past 7 days will give you a /total response for each of those days, individually
+    * All available aggregations and filters are the same as with /total
+
+
+Below are some examples of these APIs in action.
+
+_Totals request, aggregating by domain over the past 7 days:_
+
+```
+https://nightly.opencost.io/model/customCost/total?window=7d&aggregate=domain

```

Comment:
That URL doesn't map to OpenCost, should be `https://nightly.opencost.io/customCost/total?window=7d&aggregate=domain`. Is `domain` a Datadog aggregator because I haven't seen it in OpenCost?

Reply:
This request url was taken straight from the doc.... are you able to provide the correct url?

Reply:
That doc did a find and replace on `nightly.kubecost.xy`z with `nightly.opencost.io`, AFAIK there's no such instance